hathor/manager.py

```diff
@@ -165,6 +165,9 @@ def __init__(self, reactor: Reactor, peer_id: Optional[PeerId] = None, network:
             assert self.tx_storage.indexes is not None
             self.log.debug('enable utxo index')
             self.tx_storage.indexes.enable_utxo_index()
+        if event_storage is not None:
+            self.event_manager = EventManager(event_storage, self.my_peer.id or "")
```

Comment:
I feel we should create this object in the builder and just inject it here. What do you think?

Reply:
You're right. It's still an open PR. We can do it later in a refactor. Thanks!